### PR TITLE
#439: git meta add support '-u' flag

### PR DESCRIPTION
**Pull request by s-xiang**

### CODE REVIEW

File: node/test/util/add.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ a=B|x=S:C2-1 a/b=Sa:1;Oa/b W x/y/z=a,x/r/z=b;Bmaster=2`,
                 }
                 const doAdd = co.wrap(function *(repos) {
                     const repo = repos.x;
-                    yield Add.stagePaths(repo, c.paths, stageMeta);
+                    yield Add.stagePaths(repo, c.paths, stageMeta, false);
```
Comment (bpeabody) [inline node/test/util/add.js]: Please make the value of this flag come from the test case, and add a couple of cases that exercise it.

File: node/lib/util/add.js [hunk 2]
```diff
@@ -69,8 +71,16 @@ exports.stagePaths = co.wrap(function *(repo, paths, stageMetaChanges) {
             const subRepo = yield SubmoduleUtil.getRepo(repo, name);
             const workdir = subStat.workdir.status.workdir;
             const index = yield subRepo.index();
-            yield Object.keys(workdir).map(
-                filename => index.addByPath(filename));
+            yield Object.entries(workdir).map(([filename, status]) => {
+                // if -u flag is provided, update tracked files only.
+                if (update) {
+                    if (RepoStatus.FILESTATUS.ADDED != status) {
```
Comment (novalis) [inline node/lib/util/add.js]: Is DELETED a thing?

Comment (novalis) [reply node/lib/util/add.js]: Actually, I guess it's OK if DELETED is a thing.

Comment (novalis) [reply node/lib/util/add.js]: But you could combine the two if statements, I think.

File: node/test/util/add.js [hunk 3]
```diff
@@ -129,17 +129,27 @@ a=B|x=S:C2-1 a/b=Sa:1;Oa/b W x/y/z=a,x/r/z=b;Bmaster=2`,
                 paths: [],
                 stageMeta: false,
             },
+            "add only tracked files": {
+                initial: "a=B|x=U:Os W README.md=foo",
+                paths: ["s"],
+                expected: "x=E:Os I README.md=foo",
+                update: true,
+            },
```
Comment (bpeabody) [inline node/test/util/add.js]: try:
```
initial: "a=B|x=U:Os W README.md=foo,newFile=a",
expected: "x=E:Os I README.md=foo!W newFile=a",
```

**Pull request merged**